core/arch/arm/kernel/link.mk

```diff
@@ -92,9 +93,37 @@ $(link-script-pp): $(link-script) $(link-script-extra-deps)
 	$(q)$(CPPcore) -Wp,-P,-MT,$@,-MD,$(link-script-dep) \
 		$(link-script-cppflags) $< > $@
 
+define update-buildcount
+	@$(cmd-echo-silent) '  UPD     $(1)'
+	$(q)if [ ! -f $(1) ]; then \
+		mkdir -p $(dir $(1)); \
+		echo 1 >$(1); \
+	else \
+		expr 0`cat $(1)` + 1 >$(1); \
+	fi
+endef
+
+define gen-version-o
+	$(call update-buildcount,$(link-out-dir)/.buildcount)
+	@$(cmd-echo-silent) '  GEN     $(link-out-dir)/version.o'
+	$(q)echo -e "const char core_v_str[] =" \
+		"\"`git describe --always --dirty` \"" \
```

Comment:
We must make sure that it's also possible to build without having the source code tracked in git.